src/Definition.cpp

```diff
@@ -108,6 +108,18 @@ Definition::Definition(const std::vector<Expr> &args, const std::vector<Expr> &v
     }
 }
 
+Definition::Definition(bool is_init, const Expr &predicate, const std::vector<Expr> &args, const std::vector<Expr> &values,
```

Comment:
Code style nit: prefer to avoid bool args to functions like this; it's hard to read at the call site, because the semantic of what it means isn't clear. Prefer to use something like `enum class DefinitionInit { IsInit, IsNotInit};` so that the meaning at the call site is immediately obvious.

Reply:
In this case at the call sites, it's just treating the boolean as another field of the deserialized `Definition` and it doesn't have any additional semantic meaning other than that, so I'm not sure if we want to do the enum change here.

Reply:
Ah, gotcha. OK.